### DISCUSSION TITLE
Quick question about k_point and periodic boundary conditions

### DISCUSSION THREAD

**Discussion by pwflanigan** [NanoComp/meep#2818] [Q&A] [answered]
Let's say I wanted to make a system (`cell_size = (sx,sy,sz)`) periodic in the x and y directions. When I specify the k_point in the simulation object, if I set `k_point=(sx,sy,0)` then the system would repeat itself at exactly the boundary, correct? So if I did something like `k_point=(sx+1,sy,0)`, then there would be a gap of 1 length unit in between each instance of the system as I go along the x direction? 

I was looking at the example [https://github.com/NanoComp/meep/blob/master/python/examples/holey-wvg-bands.ipynb](url) and I wasn't sure why the specified k_point was set at (0.4,0,0) even though sx was 1.0.

EDIT: to clarify, I do know that you can achieve two-directional periodicity by putting PML's on just the z direction, but want to understand what exactly k_point is doing, since I was kind of confused by the manual's explanation. If my understanding is correct, what does it mean physically to set `k_point=(0,0,0)`?

Thanks for the help.

**Comment by stevengj** [comment, marked as answer]
You are making a [category error](https://en.wikipedia.org/wiki/Category_mistake) — `k_point` is not the kind of thing you think it is.

For periodic or Bloch-periodic/quasi-periodic boundary conditions, the period in each direction is always corresponding the cell size.   This has nothing to do with `k_point`.  There is never a "gap".

`k_point` specifies a *phase* relation between one period and the next.  `k_point=(0,0,0)` corresponds to ordinary periodic boundaries.  The elements of `k_point` have units of 1/distance, so `k_point=(sx,sy,0)` is nonsensical (wrong units).  From the [manual](https://meep.readthedocs.io/en/latest/Python_User_Interface/):

<img width="534" alt="image" src="https://github.com/NanoComp/meep/assets/2913679/ec85ad00-f8b4-4d5d-bbe5-c258daa01b23">
